content/history/sweepstakes.inc

```diff
@@ -1,34 +1,52 @@
 <?php
-  function getSweepstakesByYear($year){
-    $sweepstakes = new sweepstakes();
-    $sweepstakesCommittee = $sweepstakes->GetList(array(array("year","=",$year)));
-    $committeeArr = array();
-    foreach($sweepstakesCommittee as $sweepstakes){
-      $sweepstakesRole = $sweepstakes->GetSweepstakesrole();
-      $person = $sweepstakes->GetPerson();
-      $committeeArr[$sweepstakesRole->description] = $person;
-    }
-    return $committeeArr;
+  function getAllSweepstakes() {
+    $query = "SELECT DISTINCT y.year,
+                              concat(pch.firstname, ' ', pch.lastname) AS chair, pch.personid AS chairid,
+	                            concat(pac.firstname, ' ', pac.lastname) AS asst, pac.personid AS asstid,
+	                            concat(psf.firstname, ' ', psf.lastname) AS safety, psf.personid AS safetyid,
+	                            concat(phj.firstname, ' ', phj.lastname) AS judge, phj.personid AS judgeid
+                  FROM hist_sweepstakes y
+                    LEFT JOIN hist_sweepstakes ch ON ch.year = y.year AND ch.role = 'Sweepstakes Chairman'
+                      LEFT JOIN hist_people pch ON ch.personid = pch.personid
+	                  LEFT JOIN hist_sweepstakes ac ON ac.year = y.year AND ac.role = 'Assistant Chairman'
+                      LEFT JOIN hist_people pac ON ac.personid = pac.personid
+	                  LEFT JOIN hist_sweepstakes sf ON sf.year = y.year AND sf.role = 'Safety Chairman'
+                      LEFT JOIN hist_people psf ON sf.personid = psf.personid
+	                  LEFT JOIN hist_sweepstakes hj ON hj.year = y.year AND hj.role = 'Head Judge'
+                      LEFT JOIN hist_people phj ON hj.personid = phj.personid
+                  ORDER BY y.year DESC;";
+
+    global $HISTORY_DATABASE;
+    return dbQuery($HISTORY_DATABASE, $query);
   }
 
+  function getSweepstakesYear($year) {
+    // We are cheating a bit by using ORDER BY field here.  We do 10 - FIELD() to put unknown roles at the bottom and
```

Comment:
nit: Update ordering comment above since the `10 - ` bit was taken out. Also is `role` the string name of the role?

Reply:
will fix (by removing comment)

We fixed the sorting with an enum.  So, 'role' sorts like the enum (which has a sane order), but is actually providing the string (which, if you sorted by, would not have a sane order).